browse-scaler/src/cmr.js

```diff
@@ -38,47 +39,98 @@ exports.getAuthorizationToken = getAuthorizationToken;
   const response = await fetch(cmrEndpoint + conceptId, {
     method: 'GET',
     headers: {
-      'Authorization': token
+      Authorization: token
     }
   })
-      .then(res => res.json())
-      .then(json => {
-        if (json.errors) {
-          throw new Error(`The following errors occurred: ${json.errors}`);
-        } else {
-          return json.feed.entry[0];
-        }
-      })
-      .catch(error => {
-        console.log(`Could not find concept ${conceptId}: ${error}`);
-      });
+    .then(res => res.json())
+    .then(json => {
+      if (json.errors) {
+        throw new Error(`The following errors occurred: ${json.errors}`);
+      } else {
+        return json.feed.entry[0];
+      }
+    })
+    .catch(error => {
+      console.log(`Could not find concept ${conceptId}: ${error}`);
+    });
   return response;
 };
 
+/**
+ * Parse and return the array of data from the nested response body
+ * @param {Object} jsonResponse HTTP response from the CMR endpoint
+ */
+const parseJsonBody = jsonResponse => {
+  const { data } = jsonResponse;
+  const { feed } = data;
+  const { entry } = feed;
+  const [granule] = entry;
+  return granule;
+};
+
+/**
+ * Fetch a single collection from CMR search
+ * @param {String} conceptId Collection concept id from CMR
+ * @param {String} token An optional Echo Token
+ * @returns [{JSON}] An array of UMM JSON collection results
+ */
+
+const fetchCmrGranule = async conceptId => {
+  const requestHeaders = {};
+  const token = config.CMR_ECHO_TOKEN || (await getAuthorizationToken());
+  if (token) {
+    requestHeaders.Authorization = token;
+  }
+
+  let response;
+  try {
+    response = await axios({
+      url: `${config.CMR_ROOT_URL}/search/granules.json?concept_id=${conceptId}`,
+      method: 'GET',
+      headers: requestHeaders,
+      json: true,
+      timeout: config.TIMEOUT_INTERVAL
+    });
+  } catch (error) {
+    console.log(`Could not fetch granule ${conceptId} due to error: ${error}`);
+    return null;
+  }
+  return parseJsonBody(response);
+};
+
 /**
  * getBrowseImageFromConcept: Given a CMR concept, marshall the JSON and
  * filter any associated links to find browse images associated with the concept
  * @param {JSON} concept the JSON metadata associated with a CMR concept
  * @returns {String} the image url if one is found, or null if none is found
  */
-exports.getBrowseImageFromConcept = async concept => {
+exports.getBrowseImageFromConcept = async (concept, imageSrc) => {
   if (!concept) {
-    console.error ('No concept provided to getBrowseImageFromConcept');
+    console.error('No concept provided to getBrowseImageFromConcept');
     return;
   }
   try {
     const { links } = concept;
     const imgRegex = /\b(browse#)$/;
-    const imgurl = links.filter(link => imgRegex.test(link.rel))[0];
+    const imageUrls = links.filter(link => imgRegex.test(link.rel));
+    // console.debug(`🚀 links from metadata ${JSON.stringify(links)}`);
```

Comment:
These were already in the code I just added the rocket because it was hard to find them in the docker container output